### PR TITLE
Remove GetCachedStateRangeScanIterator

### PR DESCRIPTION
**Pull request by mbrandenburger**
- Remove GetCachedStateRangeScanIterator
- Badger autoclean
- Add simple badger benchmark

Signed-off-by: Marcus Brandenburger <bur@zurich.ibm.com>

### CODE REVIEW

**Comment by adecaro** [comment]
@mbrandenburger , could you already prepare a PR that updates the token-sdk given that it depends on API that this PR removes?

**Comment by mbrandenburger** [comment]
> @mbrandenburger , could you already prepare a PR that updates the token-sdk given that it depends on API that this PR removes?

Here the PR on the token-sdk. CI runs successfully https://github.com/hyperledger-labs/fabric-token-sdk/pull/311

File: platform/fabric/core/generic/vault.go [hunk 1]
```diff
@@ -66,10 +65,5 @@ func NewVault(config *config.Config, channel string, cacheSize int) (*vault.Vaul
 		return nil, nil, errors.Wrapf(err, "failed creating txid store")
 	}
 
-	if cacheSize > 0 {
```
Comment (adecaro) [inline platform/fabric/core/generic/vault.go:69]: why removing this? Without this cache we pay a huge penalty.

File: platform/view/services/db/db.go [hunk 2]
```diff
@@ -48,13 +48,6 @@ func Drivers() []string {
 	return list
 }
 
-func unregisterAllDrivers() {
```
Comment (adecaro) [inline platform/view/services/db/db.go:51]: how is this part of this PR?

File: platform/view/services/kvs/kvs.go [hunk 3]
```diff
@@ -57,39 +54,23 @@ func New(driverName, namespace string, sp view.ServiceProvider) (*KVS, error) {
 	return &KVS{
 		namespace: namespace,
 		store:     persistence,
-		cache:     map[string][]byte{},
 	}, nil
 }
 
 func (o *KVS) Exists(id string) bool {
-	// is in cache?
```
Comment (adecaro) [inline platform/view/services/kvs/kvs.go:65]: why removing this cache?

**Review by adecaro — CHANGES_REQUESTED** [review]

**Review by adecaro — APPROVED** [review]

File: platform/view/services/db/driver/badger/badger.go [hunk 4]
```diff
@@ -7,104 +7,61 @@ SPDX-License-Identifier: Apache-2.0
...
-		itemsMap:      map[string]*ItemList{},
-		itemsListSize: 20000,
-	}, nil
+	// start our auto cleaner
+	autoCleaner(db)
+
+	return &badgerDB{db: db}, nil
+}
+
+func autoCleaner(db *badger.DB) {
```
Comment (adecaro) [inline platform/view/services/db/driver/badger/badger.go:49]: What about additing a description here? Why 5 mins and 0.7?

**Review by adecaro — APPROVED** [review]

File: platform/view/services/db/driver/badger/badger.go [hunk 5]
```diff
@@ -7,104 +7,61 @@ SPDX-License-Identifier: Apache-2.0
...
+		ticker := time.NewTicker(5 * time.Minute)
+		defer ticker.Stop()
+		for range ticker.C {
+		again:
+			if db.IsClosed() {
+				// no need to clean anymore
+				return
+			}
+			err := db.RunValueLogGC(0.7)
+			if err == nil {
```
Comment (adecaro) [inline platform/view/services/db/driver/badger/badger.go:60]: does this mean that if no error happens, RunValueLogGC is called continuously? 
Should `again` come before the for?

**Review by adecaro — CHANGES_REQUESTED** [review]

**Review by adecaro — APPROVED** [review]

**Pull request merged**